yarn-project/sequencer-client/src/block_builder/solo_block_builder.ts

```diff
@@ -128,6 +129,18 @@ export class SoloBlockBuilder implements BlockBuilder {
     // Check txs are good for processing
     this.validateTxs(txs);
 
+    const emptyTxHash = await Tx.emptyTxHash();
+
+    for (const tx of txs) {
+      if (tx.hash.equals(emptyTxHash)) {
+        continue;
+      }
+      tx.data.end.newNullifiers = [
+        Fr.fromBuffer(tx.hash.buffer),
+        ...tx.data.end.newNullifiers.slice(0, KERNEL_NEW_NULLIFIERS_LENGTH - 1),
+      ];
```

Comment:
Does it make sense to throw here if the original set of nullifiers already had `KERNEL_NEW_NULLIFIERS_LENGTH` non-zero nullifiers, so we don't inadvertently drop the last one? I don't think we'll run into it, but if we do, better to fail loudly.

Reply:
Probably, yes